proto/proto/tsmt.proto

```diff
@@ -0,0 +1,38 @@
+// Message definitions related to Tiered Sparse Merkle Trees (TSMT).
+//
+// Notes about TSMT:
+// - The tree is a key value store, both are digest values.
+// - The key's most significant byte is used to deterministically assign the
+// pair to one of the tree's entries.
+// - The tree is denominated tiered because entries are only possible at
+// specific tiers, each tier corresponding to a depth (16/32/48/64).
+// - The key's most significant quibbles (16 bits), are used to determine the
+// pair's entry at a given tier.
+// - Depths 16/32/48 contain at most one value. Pair collision on these tiers
+// causes them to to be pushed to lower tiers.
+// - The depth 64 may store a list of entries. The entries are stored in sorted
+// order, determined by their keys.
+// - The tree supports non-inclusion proofs. Achieved with a merkle path and
+// the absenced of the key in the opened leaf.
+// - The tree is initialized with null word [0,0,0,0] for the leaves.
+syntax = "proto3";
+package tsmt;
+
+import "digest.proto";
+
+message Leaf {
+    digest.Digest key = 1;
+    optional digest.Digest value = 2;
```

Comment:
This is `optional` because that is how empty leaves are encoded (The leaf itself is set to `[0,0,0,0]`, here that would be set on `key`).

Reply:
I didn't quite follow this part. If the value is `[0, 0, 0, 0]`, do we basically just us a flag rather than storing all 4 zeros? Or is this something else?

Reply:
changed, now using the length of the vector as a flag instead.